keras_cv/layers/object_detection/roi_generator.py

```diff
@@ -0,0 +1,181 @@
+            nms_iou_threshold = self.nms_iou_threshold_train
+        else:
+            pre_nms_topk = self.pre_nms_topk_test
+            post_nms_topk = self.post_nms_topk_test
+            nms_score_threshold = self.nms_score_threshold_test
+            nms_iou_threshold = self.nms_iou_threshold_test
+
+        rois = []
+        roi_scores = []
+        for level in sorted(multi_level_scores.keys()):
+            scores = multi_level_scores[level]
+            boxes = multi_level_boxes[level]
+            _, num_boxes = scores.get_shape().as_list()
+            level_pre_nms_topk = min(num_boxes, pre_nms_topk)
+            level_post_nms_topk = min(num_boxes, post_nms_topk)
+            scores, sorted_indices = tf.nn.top_k(
+                scores, k=level_pre_nms_topk, sorted=True
+            )
+            boxes = tf.gather(boxes, sorted_indices, batch_dims=1)
+            # convert from input format to yxyx for the TF NMS operation
+            boxes = bounding_box.convert_format(
+                boxes,
+                source=self.bounding_box_format,
+                target="yxyx",
+            )
+            # TODO(tanzhenyu): consider supporting soft / batched nms for accl
+            selected_indices, num_valid = tf.image.non_max_suppression_padded(
+                boxes,
+                scores,
+                max_output_size=level_post_nms_topk,
+                iou_threshold=nms_iou_threshold,
+                score_threshold=nms_score_threshold,
+                pad_to_max_output_size=True,
+                sorted_input=True,
+                canonicalized_coordinates=True,
+            )
+            # convert back to input format
+            boxes = bounding_box.convert_format(
+                boxes,
+                source="yxyx",
+                target=self.bounding_box_format,
+            )
+            level_rois = tf.gather(boxes, selected_indices, batch_dims=1)
+            level_roi_scores = tf.gather(scores, selected_indices, batch_dims=1)
+            level_rois = level_rois * tf.cast(
+                tf.reshape(tf.range(level_post_nms_topk), [1, -1, 1])
+                < tf.reshape(num_valid, [-1, 1, 1]),
+                level_rois.dtype,
+            )
+            level_roi_scores = level_roi_scores * tf.cast(
+                tf.reshape(tf.range(level_post_nms_topk), [1, -1])
+                < tf.reshape(num_valid, [-1, 1]),
+                level_roi_scores.dtype,
+            )
+            rois.append(level_rois)
+            roi_scores.append(level_roi_scores)
+
+        rois = tf.concat(rois, axis=1)
+        roi_scores = tf.concat(roi_scores, axis=1)
+        _, num_valid_rois = roi_scores.get_shape().as_list()
+        overall_top_k = min(num_valid_rois, post_nms_topk)
+        roi_scores, sorted_indices = tf.nn.top_k(
+            roi_scores, k=overall_top_k, sorted=True
+        )
+        rois = tf.gather(rois, sorted_indices, batch_dims=1)
+
+        return rois, roi_scores
```

Comment:
do u need a get_config() for serialization?

Reply:
Good catch. Done.